core/trino-main/src/test/java/io/trino/execution/scheduler/TestingTaskSourceFactory.java

```diff
@@ -139,7 +141,7 @@ public List<TaskDescriptor> getMoreTasks()
                 result.add(task);
             }
 
-            return result.build();
+            return immediateFuture(result.build());
```

Comment:
It would be great to simulate loading with delay and exercise related code paths (e.g.: make sure it blocks if the future returned is note done, make sure it unblocks when it is done, etc.)